app/src/main/java/com/tatar/currencify/presentation/feature/CurrencyConversionViewModel.kt

```diff
@@ -0,0 +1,120 @@
+package com.tatar.currencify.presentation.feature
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import com.tatar.currencify.domain.feature.GetRatesTask
+import com.tatar.currencify.domain.feature.LatestRatesEntity
+import com.tatar.currencify.presentation.util.addTo
+import com.tatar.currencify.presentation.viewmodel.Event
+import io.reactivex.rxjava3.core.BackpressureStrategy
+import io.reactivex.rxjava3.disposables.CompositeDisposable
+import io.reactivex.rxjava3.kotlin.Flowables
+import io.reactivex.rxjava3.subjects.BehaviorSubject
+import javax.inject.Inject
+
+class CurrencyConversionViewModel @Inject internal constructor(
+    private val getRatesTask: GetRatesTask
+) : ViewModel() {
+
+    private val disposables = CompositeDisposable()
+
+    private val _currencyConversionResult = MutableLiveData<CurrencyConversionResult>()
+    val currencyConversionResult: LiveData<CurrencyConversionResult> get() = _currencyConversionResult
+
+    private val _loadingState = MutableLiveData<Boolean>()
+    val loadingState: LiveData<Boolean> get() = _loadingState
+
+    private val _responderClickEvent = MutableLiveData<Event<Unit>>()
+    val responderClickEvent: LiveData<Event<Unit>> get() = _responderClickEvent
+
+    private val _animationEvent = MutableLiveData<Event<Unit>>()
```

Comment:
this event is a bit vague by it's name, like what does it do with animation. hideKeyboard Event for example, is pretty clear, but I'm having a hard time to understand what animation Event does by looking at the code. Could you make it more clear maybe?

Reply:
Will refactor